src/commands/cmd_query.c

```diff
@@ -61,12 +61,20 @@ void Graph_Query(void *args) {
 
 	QueryCtx_BeginTimer(); // Start query timing.
 
+	cypher_parse_result_t *params_parse_result = parse_params(command_ctx->query);
+	if(params_parse_result == NULL) goto cleanup;
+	if(AST_Validate_QueryParams(ctx, params_parse_result) != AST_VALID) goto cleanup;
+
+	// Query caching enabler.
+	AST_Extract_Params(params_parse_result);
+	const char *query_string = AST_ExtractQueryString(params_parse_result);
+
 	// Parse the query to construct an AST.
-	cypher_parse_result_t *parse_result = parse(command_ctx->query);
+	cypher_parse_result_t *parse_result = parse_query(query_string);
```

Comment:
Should initialize and NULL-set `parse_result` before here, as we could error on `AST_Validate_QueryParams` and jump to the cleanup routine before setting it here otherwise.